tree/ntuple/v7/inc/ROOT/RColumnElement.hxx

```diff
@@ -583,6 +584,39 @@ public:
    void Unpack(void *dst, void *src, std::size_t count) const final;
 };
 
+template <>
+class RColumnElement<float, EColumnType::kReal16> : public RColumnElementBase {
+public:
+   static constexpr bool kIsMappable = false;
+   static constexpr std::size_t kSize = sizeof(float);
+   static constexpr std::size_t kBitsOnStorage = 16;
+   RColumnElement() : RColumnElementBase(kSize) {}
+   bool IsMappable() const final { return kIsMappable; }
+   std::size_t GetBitsOnStorage() const final { return kBitsOnStorage; }
+
+   void Pack(void *dst, void *src, std::size_t count) const final
+   {
+      float *floatArray = reinterpret_cast<float *>(src);
+      std::uint16_t *uint16Array = reinterpret_cast<std::uint16_t *>(dst);
+
+      for (std::size_t i = 0; i < count; ++i) {
+         uint16Array[i] = Internal::FloatToHalf(floatArray[i]);
+         ByteSwapIfNecessary(uint16Array[i]);
+      }
+   }
+
+   void Unpack(void *dst, void *src, std::size_t count) const final
+   {
+      float *floatArray = reinterpret_cast<float *>(dst);
+      std::uint16_t *uint16Array = reinterpret_cast<std::uint16_t *>(src);
+
+      for (std::size_t i = 0; i < count; ++i) {
+         floatArray[i] = Internal::HalfToFloat(uint16Array[i]);
+         ByteSwapIfNecessary(floatArray[i]);
```

Comment:
Shouldn't these two lines be reversed (in reverse order to packing)?

Reply:
Yes, you're completely right!